src/utils/session.ts

```diff
@@ -198,12 +218,21 @@ export async function unsealSession(
  */
 export function clearSession(
   event: H3Event,
-  config: Partial<SessionConfig>,
+  config: SessionConfig,
 ): Promise<void> {
-  const sessionName = config.name || DEFAULT_SESSION_NAME;
-  if (event.context.sessions?.[sessionName]) {
-    delete event.context.sessions![sessionName];
+  const sessionName = getSessionName(config);
+  if (!event.context.sessions) {
+    event.context.sessions = new EmptyObject();
   }
+
+  // Clobber the original session with a new session
+  event.context.sessions![sessionName] = {
```

Comment:
This should be reverted to allow this PR move forward as a refactor step.

Reply:
I'm a bit confused @pi0. This line is the main point of this PR. If you want to isolate changes I'd be more apt to revert the refactor changes on this PR and apply them on a separate PR intended for refactor changes. Thoughts?

Reply:
We can do this as part of #1022 for session rotate. clear purpose is to destroy session fully not to create a new one.

Reply:
@pi0, I'm in complete agreement on that point, but the implementation at the code level is not 1:1 with the user experience. When the user calls `clear()` they want the session cleared, and then if they call `update()` they would expect that this creates a new session (or at least that's what I would assume). Because the `update()` call is eagerly restoring the session from the `request`, calling `update()` returns a mutation of the same session that was previously cleared.

**TLDR;** If a new session is not set on the `context.sessions[sessionName]` when `clear()` is called, then the session isn't actually called, it's just temporarily reset (and will be restored on next access call).